go-controller/pkg/libovsdb/libovsdb.go

```diff
@@ -108,10 +115,22 @@ func newClientWithMonitor(cfg config.OvnAuthConfig, stopCh <-chan struct{}, dbMo
 		cancel()
 	}()
 
-	_, err = c.MonitorAll(ctx)
-	if err != nil {
-		c.Close()
-		return nil, err
+	defer func() {
+		if err != nil {
+			c.Close()
+		}
+	}()
+
+	if len(tables) == 0 {
+		if _, err = c.MonitorAll(ctx); err != nil {
+			return nil, err
+		}
+	} else {
+		for _, table := range tables {
+			if _, err = c.Monitor(ctx, c.NewMonitor(client.WithTable(table))); err != nil {
```

Comment:
@dave-tucker : the monitor API changed recently and I had to update this from `NewTableMonitor`. Could you have a quick look here please and let me know if you feel this is appropriate usage of this API?

Reply:
No that doesn't look right. You're setting up a dedicated monitor per table instead of one monitor for many tables.

You'd want to do something like this:

```go
var options []client.MonitorOption
for _, table := range tables {
  options = append(options. client.WithTable(table))
}
c.Monitor(options...)
```

Reply:
Thanks for taking a look!